magefile.go

```diff
@@ -8,31 +8,149 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"os"
 	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
 
-	//mage:import
 	"github.com/elastic/elastic-agent-libs/dev-tools/mage"
+	devtools "github.com/elastic/elastic-agent-shipper/dev-tools/common"
+	"github.com/elastic/elastic-agent-shipper/tools"
+
+	//mage:import
+
 	"github.com/elastic/elastic-agent-libs/dev-tools/mage/gotool"
 	"github.com/magefile/mage/mg"
 	"github.com/magefile/mage/sh"
 )
 
-var (
-	goLicenserRepo = "github.com/elastic/go-licenser@v0.4.1"
+const (
+	GoreleaserRepo = "github.com/goreleaser/goreleaser@v1.6.3"
 )
 
 // Aliases are shortcuts to long target names.
 // nolint: deadcode // it's used by `mage`.
 var Aliases = map[string]interface{}{
-	"llc":  mage.Linter.LastChange,
-	"lint": mage.Linter.All,
+	"build":           Build.Binary,
+	"unitTest":        Test.Unit,
+	"integrationTest": Test.Integration,
+}
+
+// BUILD
+
+// Build contains targets related to linting the Go code
+type Build mg.Namespace
+
+// All builds binaries for the all os/arch.
+func (Build) All() {
+	os.Setenv("PLATFORM", "all")
+	mg.Deps(Build.Binary)
 }
 
-func Build() {
-	sh.Run("go", "build")
+// Clean removes the build directory.
+func (Build) Clean(test string) {
+	os.RemoveAll("build") // nolint:errcheck //not required
 }
 
-// Check runs all the checks
+// CheckBinaries checks if the binaries are generated (for now).
+func (Build) CheckBinaries() error {
+	path := filepath.Join("build", "binaries")
+	for _, platform := range devtools.PlatformFiles {
+		var execName = "elastic-agent-shipper"
+		if strings.Contains(platform, "windows") {
+			execName += ".exe"
+		}
+		binary := filepath.Join(path, fmt.Sprintf("%s-%s-%s", "elastic-agent-shipper", tools.DefaultBeatVersion, platform), execName)
+		if _, err := os.Stat(binary); os.IsNotExist(err) {
```

Comment:
if there's other error we assume it's existing, I wonder if we should rather be more pesimistic

Reply:
I've added an overall check